api/src/main/java/com/velocitypowered/api/command/RawCommand.java

```diff
@@ -29,15 +30,21 @@ default void execute(CommandSource source, String @NonNull [] args) {
    * @param currentLine the current, partial command line for this command
    * @return tab complete suggestions
    */
-  default List<String> suggest(CommandSource source, String currentLine) {
-    return ImmutableList.of();
+  default CompletableFuture<List<String>> suggest(CommandSource source, String currentLine) {
+    return CompletableFuture.completedFuture(ImmutableList.of());
   }
 
   @Override
-  default List<String> suggest(CommandSource source, String @NonNull [] currentArgs) {
+  default CompletableFuture<List<String>> suggestAsync(CommandSource source,
+      String @NonNull [] currentArgs) {
     return suggest(source, String.join(" ", currentArgs));
   }
 
+  @Override
+  default List<String> suggest(CommandSource source, String @NonNull [] currentArgs) {
```

Comment:
I would deprecate this method/rename the other one to avoid confusing API clients. How do they know which of the 3 methods should they use?

Reply:
If I had to be completely honest, it's probably better to deprecate `Command` and split it into `RawCommand` and `RegularCommand`, perhaps also introducing `CommandContext` and `CommandExecutionContext` interface at the same time, with `CommandContext` providing command metadata and `CommandExecutionContext` focusing on the invocation of a command.

My 2 cents.